src/CPUSolver.cpp

```diff
@@ -819,14 +819,36 @@ void CPUSolver::transferBoundaryFlux(int track_id,
   /* For the "forward" direction */
   if (direction) {
     start = _tracks[track_id]->isNextOut() * _polar_times_groups;
-    bc = std::min((int)_tracks[track_id]->getBCOut(), 1);
+
+    /* If the outgoing boundary condition is vacuum, set bc to 0 indicating the
+     * track flux of the outgoing track will be zeroed out */
```

Comment:
Are you sure you have to cast them? I thought you could simply multiply a float by a boolean, but I may be wrong (def worthwhile to make sure both GNU and Intel allow for this). If it is not allowed, yes, I think that your proposal is a good compromise.